sdk/nodejs/tests/automation/localWorkspace.spec.ts

```diff
@@ -1487,3 +1586,129 @@ const normalizeConfigKey = (key: string, projectName: string) => {
     }
     return "";
 };
+
+/**
+ * Augments the provided {@link LocalWorkspaceOptions} so that they reference a
+ * either a file backend or a cloud backend, depending on whether PULUMI_ACCESS_TOKEN
+ * is set in the environment.
+ */
+function withTestBackend(
+    opts?: LocalWorkspaceOptions,
+    name?: string,
+    description?: string,
+    runtime?: string,
+): LocalWorkspaceOptions {
+    if (process.env.PULUMI_ACCESS_TOKEN) {
+        return withCloudBackend(opts, name, description, runtime);
+    }
+    return withTemporaryFileBackend(opts, name, description, runtime);
+}
+
+function withCloudBackend(
+    opts?: LocalWorkspaceOptions,
+    name?: string,
+    description?: string,
+    runtime?: string,
+): LocalWorkspaceOptions {
+    const backend = {
+        url: "https://api.pulumi.com",
+    };
+    if (name === undefined) {
+        name = "node_test";
+    }
+    if (runtime === undefined) {
+        runtime = "nodejs";
+    }
+    return {
+        ...opts,
+        projectSettings: {
+            // We are obliged to provide a name and runtime if we provide project
+            // settings, so we do so, but we spread in the provided project settings
+            // afterwards so that the caller can override them if need be.
+            name: name,
+            runtime: runtime as ProjectRuntime,
+            description: description,
+
+            ...opts?.projectSettings,
+            backend,
+        },
+    };
+}
+
+function withTemporaryFileBackend(
+    opts?: LocalWorkspaceOptions,
+    name?: string,
+    description?: string,
+    runtime?: string,
+): LocalWorkspaceOptions {
+    console.log("with file backend", name, description);
+    const tmpDir = tmp.dirSync({
+        prefix: "nodejs-tests-automation-",
+        unsafeCleanup: true,
+    });
+
+    const backend = { url: `file://${tmpDir.name}` };
+
+    if (name === undefined) {
+        name = "node_test";
+    }
+    if (runtime === undefined) {
+        runtime = "nodejs";
+    }
+
+    return withTestConfigPassphrase({
+        ...opts,
+        pulumiHome: tmpDir.name,
+        projectSettings: {
+            // We are obliged to provide a name and runtime if we provide project
+            // settings, so we do so, but we spread in the provided project settings
+            // afterwards so that the caller can override them if need be.
+            name: name,
+            runtime: runtime as ProjectRuntime,
+            description: description,
+
+            ...opts?.projectSettings,
+            backend,
+        },
+    });
+}
+
+function withTemporaryFileBackendTmp(opts?: LocalWorkspaceOptions): LocalWorkspaceOptions {
```

Comment:
This is probably not needed anymore, I assume it was this for testing using `/tmp` vs what `tmp.dirSync` uses by default?